### PR TITLE
5634 unit testing

### PR DESCRIPTION
**Pull request by rschlaefli**
This PR adds some initial unit tests to the following classes/methods:

- `FileMetadata.getFileDateToDisplay`
- `DataAccess.createNewStorageIO (partial)`
- `FileSizeChecker.testIsAllowedFileSize`

I also added and used the testing dependency on `junit.jupiter.params` as described in #5657.

## Related Issues

- connects to issue #5634 (as well as PRs #5642 and #5643)

## Pull Request Checklist

- [x] Unit [tests][] completed
- [x] Integration [tests][]: None
- [x] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [x] [Documentation][docs] completed
- [x] Merged latest from "develop" [branch][] and resolved conflicts

### CODE REVIEW

File: src/test/java/edu/harvard/iq/dataverse/dataaccess/DataAccessTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+import java.io.IOException;
+
+
+public class DataAccessTest {
+  DataFile datafile;
+  Dataset dataset;
+  DataAccess dataAccess;
```
Comment (matthew-a-dunlap) [inline src/test/java/edu/harvard/iq/dataverse/dataaccess/DataAccessTest.java]: unused

File: src/test/java/edu/harvard/iq/dataverse/dataaccess/DataAccessTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  @Test
+  void testCreateNewStorageIO_createsFileAccessIObyDefault() throws IOException {
+    StorageIO<Dataset> storageIo = DataAccess.createNewStorageIO(dataset, "valid-tag");
+    assertTrue(storageIo.getClass().equals(FileAccessIO.class));
+  }
+
+  /*
+  TODO: testing these branches requires the possibility to inject swift and s3 client mocks
```
Comment (matthew-a-dunlap) [inline src/test/java/edu/harvard/iq/dataverse/dataaccess/DataAccessTest.java]: Injecting S3/Swift mocks would definitely be useful, but we shouldn't hold onto these tests as there is no roadmap to add the mocking.

File: src/test/java/edu/harvard/iq/dataverse/datasetutility/FileSizeCheckerTest.java [hunk 3]
```diff
@@ -6,19 +6,45 @@
...
 /**
  *
  * @author oscardssmith
  */
 public class FileSizeCheckerTest {
 
+    private FileSizeChecker fileSizeChecker;
+
+    @BeforeEach
+    public void setUp() {
```
Comment (matthew-a-dunlap) [inline src/test/java/edu/harvard/iq/dataverse/datasetutility/FileSizeCheckerTest.java]: I'd prefer if this override is just done in each method its used. You are already doing one method override and moving this into the other two where it is relevant will make things less confusing if anyone adds more tests.

File: src/test/java/edu/harvard/iq/dataverse/datasetutility/FileSizeCheckerTest.java [hunk 4]
```diff
@@ -35,4 +61,38 @@ public void testBytesToHumanReadable() {
...
+    public void testIsAllowedFileSize_allowsOnUnboundedFileSize() {
+        // initialize a system config and instantiate a file size checker
+        // ensure that a max filesize is not set
+        FileSizeChecker unboundedFileSizeChecker = new FileSizeChecker(new SystemConfig() {
+            @Override
+            public Long getMaxFileUploadSize() {
+                return null;
+            }
+        });
+        FileSizeResponse response = unboundedFileSizeChecker.isAllowedFileSize(1000L);
```
Comment (matthew-a-dunlap) [inline src/test/java/edu/harvard/iq/dataverse/datasetutility/FileSizeCheckerTest.java]: Maybe do one more of these checks with a much larger number?

File: src/test/java/edu/harvard/iq/dataverse/dataaccess/DataAccessTest.java (hunk 2 above)
Comment (rschlaefli) [reply src/test/java/edu/harvard/iq/dataverse/dataaccess/DataAccessTest.java]: Alright, I have removed these!

File: src/test/java/edu/harvard/iq/dataverse/datasetutility/FileSizeCheckerTest.java (hunk 3 above)
Comment (rschlaefli) [reply src/test/java/edu/harvard/iq/dataverse/datasetutility/FileSizeCheckerTest.java]: Okay, I moved them directly into the test cases.

File: src/test/java/edu/harvard/iq/dataverse/datasetutility/FileSizeCheckerTest.java (hunk 4 above)
Comment (rschlaefli) [reply src/test/java/edu/harvard/iq/dataverse/datasetutility/FileSizeCheckerTest.java]: I converted this to a parametrized test with 0L, 1000L, and Long.MAX_VALUE, which should cover the domain applicable to the test.

**Comment by rschlaefli** [comment]
Hi @matthew-a-dunlap! Thanks for the comments, I have resolved them in my last two commits :)

**Pull request merged**